src/network/networkd-dhcp4.c

```diff
@@ -1593,14 +1634,19 @@ int config_parse_dhcp_deny_listed_ip_address(
                 r = in_addr_from_string(AF_INET, n, &ip);
                 if (r < 0) {
                         log_syntax(unit, LOG_ERR, filename, line, r,
-                                   "DHCP deny-listed IP address is invalid, ignoring assignment: %s", n);
+                                   "DHCP '%s=' IP address is invalid, ignoring assignment: %s", lvalue, n);
                         continue;
                 }
 
-                r = set_ensure_put(&network->dhcp_deny_listed_ip, NULL, UINT32_TO_PTR(ip.in.s_addr));
+                if (STR_IN_SET(lvalue, "DenyList", "BlackList"))
+                        r = set_ensure_put(&network->dhcp_deny_listed_ip, NULL, UINT32_TO_PTR(ip.in.s_addr));
```

Comment:
I'd prefer to figure out the object to operate on first:
```C
Set **s = STR_IN_SET(lvalue, "DenyList", "BlackList") ? &network->dhcp_deny_listed_ip : &network->dhcp_allow_listed_ip;
r = set_ensure_put(s, NULL, UINT32_TO_PTR(ip.in.s_addr));
```

Reply:
Then for logging you need to compare again to know it's `DenyList` or `AllowList`